go/chat/boxer.go

```diff
@@ -413,18 +414,28 @@ func (b *Boxer) checkInvariants(ctx context.Context, convID chat1.ConversationID
 		return NewPermanentUnboxingError(err)
 	}
 
-	// Check that message type on the client header matches the body
 	body := unboxed.MessageBody
 	if !body.IsNil() {
 		bodyTyp, err := body.MessageType()
 		if err != nil {
 			return NewPermanentUnboxingError(err)
 		}
+
+		// Check that message type on the client header matches the body
 		if unboxed.ClientHeader.MessageType != bodyTyp {
 			err := fmt.Errorf("client header message type does not match body: %v(header) != %v(body)",
 				unboxed.ClientHeader.MessageType, bodyTyp)
 			return NewPermanentUnboxingError(err)
 		}
+		// Check that txID on the client header matches the body, if relevant
+		if bodyTyp == chat1.MessageType_SENDPAYMENT {
+			bodyTxID := stellar1.TransactionIDFromPaymentID(body.Sendpayment__.PaymentID)
```

Comment:
reason to use TxID is because that is what the PaymentDetails RPC method takes in